File: dmoj/dpa/dpa.cpp
#include <bits/stdc++.h>
using namespace std;

const int MN = 1e5 + 5;
int n;
int arr[MN];
int dp[MN];

int main(){
    cin >> n;
    for(int i = 0; i < n; i++) cin >> arr[i];
    for(int i = 1; i < n; i++){
        dp[i] = dp[i-1] + abs(arr[i] - arr[i-1]);
        if(i >= 2) dp[i] = min(dp[i], dp[i-2] + abs(arr[i] - arr[i-2]));
    }
    cout << dp[n-1] << '\n';
}

File: dmoj/dpa/dpa.py
import sys
input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().split()))

dp = [0]*n
for i in range(1, n):
    if i == 1:
        dp[i] = dp[i-1] + abs(arr[i]-arr[i-1])
    else:
        dp[i] = min(dp[i-1] + abs(arr[i]-arr[i-1]), dp[i-2] + abs(arr[i]-arr[i-2]))

print(dp[n-1])
